.travis.yml

```diff
@@ -7,9 +7,65 @@ python:
     - 3.6
     - pypy
     - pypy3
+matrix:
+    include:
+        # It's important to use 'macpython' builds to get the least
+        # restrictive wheel tag. It's also important to avoid
+        # 'homebrew 3' because it floats instead of being a specific version.
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='macpython 2.7'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='macpython 3.4'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='macpython 3.5'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='macpython 3.6.0'
+        - services:
+            - docker
+          env: DOCKER_IMAGE=quay.io/pypa/manylinux1_x86_64
+          before_install:
+            - bash .manylinux.sh
+            - exit 0
+        - services:
+            - docker
+          env:
+            - DOCKER_IMAGE=quay.io/pypa/manylinux1_i686
+            - PRE_CMD=linux32
+          before_install:
+            - bash .manylinux.sh
+            - exit 0
+before_install:
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then git clone https://github.com/MacPython/terryfy; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then source terryfy/travis_tools.sh; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then get_python_environment $TERRYFY_PYTHON venv; fi
 install:
+    - pip install -U pip setuptools
+    - pip install -U coveralls coverage
     - pip install .
 script:
-    - python setup.py -q test -q
+    - coverage run setup.py test -q
 notifications:
     email: false
+after_success:
+    - coveralls
+    - echo [distutils]                                  > ~/.pypirc
+    - echo index-servers = pypi                        >> ~/.pypirc
+    - echo [pypi]                                      >> ~/.pypirc
+    - echo repository=https://pypi.python.org/pypi     >> ~/.pypirc
```

Comment:
```
Note: you are uploading to the old upload URL. It's recommended to use the new URL "https://upload.pypi.org/legacy/" or to leave the URL unspecified and allow twine to choose.
Uploading zodbpickle-1.0-cp27-cp27m-macosx_10_6_intel.whl
HTTPError: 410 Client Error: Gone (This API has been deprecated and removed from legacy PyPI in favor of using the APIs available in the new PyPI.org implementation of PyPI (located at https://pypi.org/). For more information about migrating your use of this API to PyPI.org, please see https://packaging.python.org/guides/migrating-to-pypi-org/#uploading. For more information about the sunsetting of this API, please see https://mail.python.org/pipermail/distutils-sig/2017-June/030766.html) for url: https://pypi.python.org/pypi
```

I think Twine should know the right upload URL to use, so how about let's remove the `echo repository=...` override from .pypirc here (and in master)?

Reply:
And, yes, we ought to do that from every other zopefoundation project that uploads wheels from Travis:

```
mg@platonas: ~/src/zopefoundation $ grep -l twine */.travis.yml
BTrees/.travis.yml
persistent/.travis.yml
zope.index/.travis.yml
zope.interface/.travis.yml
```

Reply:
@mgedmin Thank you for digging this up. I remove the line from each of those config files.